sunpy/timeseries/metadata.py

```diff
@@ -0,0 +1,366 @@
+        # Find all results with suitable timerange.
+        results_indexes = []
+        for i in range(0, len(self.metadata)):
+            if dt in self.metadata[i][0] or not(dt):
+                results_indexes.append(i)
+        
+        # Filter out only those with the correct column.
+        results = []
+        for i in results_indexes:
+            if (colname in self.metadata[i][1]) or (not colname):
+                if indexes:
+                    results.append(i)
+                else:
+                    results.append(self.metadata[i][2])
+        
+        return results
+    
+    def get_index(self, index):
+        """
+        Return the dictionary entry at the given index.
+        
+        Parameters
+        ----------
+        index : `int`
+            The integer index of the metadata entry in the list.
+    
+        Returns
+        -------
+        metadata : `~sunpy.util.metadata.MetaDict`
+            An ordered Dictionary containing the metadata at the given index.
+        """
+        return self.metadata[index][1]
+
+    def get(self, key, default=None, **kwargs):
+        """
+        Return a list of all the matching entries in the metadata.
+        
+        Parameters
+        ----------
+        key : `str`
+            The Key to be searched in the dictionary.
+
+        default : optional
+            The Value to be returned in case key does not exist.
+
+        datetime : `str` or `~datetime.datetime` optional
+            The string (parsed using the `~sunpy.time.parse_time`) or datetime
+            that you need metadata for.
+        
+        colname : `str` optional
+            A string that can be used to narrow results to specific columns.
+
+        itemised : `bool` optional
+            Option to allow the return of the time ranges and column names (as list) that match each geven value.
+            
+        Returns
+        -------
+        list : `list`
+            Returns a list of the matching entries or the default value.
+        """
+        # Parameters
+        itemised = kwargs.get('itemised', False)
+        datetime = kwargs.get('datetime', None)
+        colname = kwargs.get('colname', None)
+        
+        # Find all matching metadata entries
+        indexes = self.find(datetime=datetime, colname=colname, indexes=True)
+        
+        # Now find each matching entry
+        results = []
+        for i in indexes:
+            # Get the value for the entry in this metadata entry
+            value = self.metadata[i][2].get(key)            
+            
+            # Need more details if itemised.
+            if itemised:
+                tr       = self.metadata[i][0]
+                colnames = self.metadata[i][1]
+                
+                # Append a tuple of these values
+                results.append(( tr, colnames, value ))
+            else:
+                # Add to the list if a result was returned
+                if value != None:

```

Comment:
`if value` is better than `value is not None` which is better than `value != None`